app/services/proofing/aamva/request/verification_request.rb

```diff
@@ -140,6 +140,20 @@ def user_provided_data_map
           }
         end
 
+        def state_id_number
+          south_carolina_zero_padding
+
+          applicant.state_id_data.state_id_number
+        end
+
+        def south_carolina_zero_padding
+          return unless applicant.state_id_data.state_id_jurisdiction == 'SC'
+
+          while applicant.state_id_data.state_id_number.length < 8
+            applicant.state_id_data.state_id_number = "0#{applicant.state_id_data.state_id_number}"
+          end
+        end
+
```

Comment:
Are we intending to mutate the original applicant value?

If yes, should we add test coverage, and consider doing this operation earlier before it reaches the proofer request?

If no, can we simplify the logic and avoid assigning to the applicant?

```suggestion
        def state_id_number
          if applicant.state_id_data.state_id_jurisdiction == 'SC'
            applicant.state_id_data.state_id_number.rjust(8, '0')
          else
            applicant.state_id_data.state_id_number
          end
        end
```

Reply:
Thanks, done in 0acd6849b